x/distribution/keeper/delegation.go

```diff
@@ -79,7 +86,7 @@ func (k Keeper) calculateDelegationRewards(ctx sdk.Context, val sdk.Validator, d
 				if endingPeriod > startingPeriod {
 					rewards = rewards.Add(k.calculateDelegationRewardsBetween(ctx, val, startingPeriod, endingPeriod, stake))
 					// note: necessary to truncate so we don't allow withdrawing more rewards than owed
-					stake = stake.MulTruncate(sdk.OneDec().Sub(event.Fraction))
+					stake = stake.MulTruncate(sdk.OneDec().Sub(event.Fraction)).MulTruncate(roundingCorrectionFactor)
```

Comment:
Should we multiply by the correction factor for *every* slash instead of just once in the final calculation?

Reply:
Well we just have to make sure that the rewards are properly accounted for - so if the stake we feed into the rewards calculation is uncorrected then we run the risk of dishing out too many rewards... so AFAICT we should probably perform this every round

Reply:
Will it even work though? Couldn't the correction factor need to be different for intermediary stake values (which we no longer have the delegator share exchange rate for)?

Reply:
See my proposed solution below to remove the use of the correction factor